### PR TITLE
Adding Go selectors page

### PR DESCRIPTION
**Pull request by flossypurse**

### CODE REVIEW

File: docs/go-selectors.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+title: Go Selectors
+sidebar_label: Selectors
+---
+
+## Overview
+
+In Go, the `select` statement lets a goroutine wait on multiple communication operations.
+A `select` **blocks until one of its cases can run**, then it executes that case.
+It chooses one at random if multiple are ready.
+Temporal's Go SDK `Selector`s are similar, but can block on sending and receiving from `Channel`s as well as `Future` deferred work.
```
Comment (mfateev) [inline docs/go-selectors.md:12]: I would rewrite this paragraph to emphasize that Go select cannot be used in Temporal workflows due to determinism. That's why Selector is provided as a replacement. As a bonus, it has additional features like the ability to listen on a Future.

File: docs/go-selectors.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	})
+```
+
+Merely matching on the channel doesn't consume the message; it has to be explicitly consumed with a `c.Receive(ctx, &signalVal)` call.
+
+## Querying Selector State
+
+TODO: document `selector.HasPending`
+Context: This was added to ensure that signals are not lost when Workflow closed.
+Explain how to ensure signals are not lost.
```
Comment (sw-yx) [inline docs/go-selectors.md:103]: any thoughts on whether/how to document this?

File: docs/go-selectors.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+---
+id: go-selectors
+title: Go Selectors
```
Comment (sw-yx) [inline docs/go-selectors.md:3]: possible alternative: "Go SDK Selectors"? to make it clear its our thing, not a Go thing

File: docs/go-selectors.md (hunk 2 above)
Comment (flossypurse) [reply docs/go-selectors.md:103]: Maybe, but right now its making this pr stale -

Comment (flossypurse) [reply docs/go-selectors.md]: Made a task - lets tackle this in another iteration -
What else needs to happen to get this into the docs?

Comment (sw-yx) [reply docs/go-selectors.md]: i dont see anything else blocking IMO

**Review by sw-yx — APPROVED** [review]

**Pull request merged**